File: Editorial/0000-0100/0053_Maximum_Subarray/CSharp/1_Optimized_Brute_Force.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        int max_subarray = int.MinValue;
        for (int i = 0; i < nums.Length; i++) {
            int current_subarray = 0;
            for (int j = i; j < nums.Length; j++) {
                current_subarray += nums[j];
                max_subarray = Math.Max(max_subarray, current_subarray);
            }
        }

        return max_subarray;
    }
}

File: Editorial/0000-0100/0053_Maximum_Subarray/Go/1_Optimized_Brute_Force.go
func maxSubArray(nums []int) int {
    max_subarray := -(int(^uint(0) >> 1)) - 1 // min int value
    for i := 0; i < len(nums); i++ {
        current_subarray := 0
        for j := i; j < len(nums); j++ {
            current_subarray += nums[j]
            if current_subarray > max_subarray {
                max_subarray = current_subarray
            }
        }
    }
    return max_subarray
}
